lms/djangoapps/grades/new/subsection_grade.py

```diff
@@ -248,13 +264,14 @@ def update(self, subsection, block_structure=None):
         if not PersistentGradesEnabledFlag.feature_enabled(self.course.id):
             return
 
-        block_structure = self._get_block_structure(block_structure)
-        subsection_grade = SubsectionGrade(subsection, self.course)
-        subsection_grade.compute(self.student, block_structure, self._scores_client, self._submissions_scores)
+        with persistence_safe_fallback():
+            block_structure = self._get_block_structure(block_structure)
```

Comment:
Putting as much outside the contextmanager is useful, so we don't have to hold the transaction open longer than necessary.